main.cpp

```diff
@@ -38,13 +38,20 @@ T::T(int v, const char* n) : value(v), name(n) {}
 
 struct Compare                                
 {
-    const T* compare(const T& a, const T& b) 
+    T* compare(T& a, T& b) 
     {
-        const T* ptr {nullptr};
+        T* ptr {nullptr};
         
-        if( a.value < b.value ) return (ptr = &a);
-        if( a.value > b.value ) return (ptr = &b);
-
+        if( a.value < b.value ) 
+        {
+            ptr = &a;
+            return ptr;
+        }
+        if( a.value > b.value ) 
+        {
+            ptr = &b;
+            return ptr;
+        }
```

Comment:
Do not create a local variable in this function.
It is not needed.
```
            if( a.value < b.value ) return the address of 'a'
            if( a.value > b.value ) return the address of 'b'
            
            return a nullptr
```
This is all you need to do.
You are currently overcomplicating the assignment.
- [ ] revise